cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/ADLSBrowser/index.js

```diff
@@ -0,0 +1,543 @@
+import {
+  setError,
+  setADLSLoading,
+  goToADLSfilePath,
+  trimSuffixSlash,
+} from 'components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/ActionCreator';
+import T from 'i18n-react';
+import orderBy from 'lodash/orderBy';
+import LoadingSVGCentered from 'components/LoadingSVGCentered';
+import isEmpty from 'lodash/isEmpty';
+import DataPrepStore from 'components/DataPrep/store';
+import lastIndexOf from 'lodash/lastIndexOf';
+import isNil from 'lodash/isNil';
+import DataprepBrowserTopPanel from 'components/DataPrep/DataPrepBrowser/DataPrepBrowserTopPanel';
+import { ConnectionType } from 'components/DataPrepConnections/ConnectionType';
+import DataPrepBrowserPageTitle from 'components/DataPrep/DataPrepBrowser/PageTitle';
+import FilePath from 'components/FileBrowser/FilePath';
+import history from 'services/history';
+
+const BASEPATH = '/';
+const PREFIX = 'features.ADLSBrowser';
+
+export default class ADLSBrowser extends Component {
+  state = {
+    contents: [],
+    path: '',
+    statePath: objectQuery(this.props, 'match', 'url') || '',
+    loading: true,
+    search: '',
+    sort: 'name',
+    sortOrder: 'asc',
+    searchFocus: true,
+    connectionId: DataPrepBrowserStore.getState().adls.connectionId,
+  };
+
+  static defaultProps = {
+    enableRouting: true,
+    scope: false,
+    browserTitle: T.translate(`${PREFIX}.TopPanel.selectData`),
+    allowSidePanelToggle: true,
+  };
+
+  static propTypes = {
+    allowSidePanelToggle: PropTypes.bool,
+    location: PropTypes.object,
+    match: PropTypes.object,
+    initialDirectoryPath: PropTypes.string,
+    noState: PropTypes.bool,
+    toggle: PropTypes.func.isRequired,
+    enableRouting: PropTypes.bool,
+    onWorkspaceCreate: PropTypes.func,
+    scope: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
+    browserTitle: PropTypes.string,
+  };
+
+  parsePath() {
+    this._isMounted = true;
+    if (!this.props.enableRouting) {
+      let path = this.getFilePath();
+      this.dataprepSubscription = DataPrepStore.subscribe(() => {
+        let path = this.getFilePath();
+        if (!isNil(path) && path !== this.state.path) {
+          goToADLSfilePath(path);
+        }
+      });
+      goToADLSfilePath(path);
+    } else {
+      this.fetchADLSDirectory(this.props);
+    }
+  }
+
+  componentDidMount() {
+    this.parsePath();
+    this.browserStoreSubscription = DataPrepBrowserStore.subscribe(() => {
+      let { adls, activeBrowser } = DataPrepBrowserStore.getState();
+      if (activeBrowser.name !== ConnectionType.ADLS) {
+        return;
+      }
+
+      if (this._isMounted) {
+        this.setState({
+          contents: adls.contents,
+          info: adls.info,
+          connectionId: adls.connectionId,
+          loading: adls.loading,
+          path: adls.path,
+          search: adls.search,
+        });
+      }
+    });
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
```

Comment:
@avantika3667 Since ADLSBrowser is almost a replica of the file browser we have this problem where when we are in a ADLS connection and if we delete the connection the user is prompted with a incorrect message of "Unable to find :connection-name:" error banner.

This will be confusing as the user just deleted the connection. The reason it is happening is because,
- When we delete a connection we redirect UI to the `/ns/default/connections` url
- At this point UI again fetches the list of connection types and connections and determines the default connection to fallback to.
- BUT the adls connection that the user just deleted makes ADLSBrowser as still the "Active browser" based on the state we maintain.
- Since the ADLS is still the active browser the ADLSBrowser component again gets mounted again (with store still having information about the connection).
- Which then makes the explore call to wrangler upon which the UI receives "Unable to find :connection-name:" error.

Right now there is a bigger problem that is waiting to explode but can you add these two statements here in the unmount handler?
```
import {
  setError,
  setADLSLoading,
  goToADLSfilePath,
  trimSuffixSlash,
  resetADLSBrowser,
} from 'components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/ActionCreator';
import { setActiveBrowser } from 'components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/Actions/commons';
....
componentWillUnmount()  {
resetADLSBrowser();
setActiveBrowser({ name: '' });
...
}
```

I have attached a gif for your reference so that you can reproduce this locally to check what I am talking about.
![deleteadls](https://user-images.githubusercontent.com/1452845/60126598-36d45980-9743-11e9-9321-e2272d365e54.gif)